ast/term.go

```diff
@@ -1810,6 +1816,161 @@ func ObjectTerm(o ...[2]*Term) *Term {
 	return &Term{Value: NewObject(o...)}
 }
 
+func LazyObject(blob map[string]interface{}) Object {
+	return &lazyObj{native: blob}
+}
+
+type lazyObj struct {
+	strict Object
+	native map[string]interface{}
+}
+
+func (l *lazyObj) force() Object {
+	if l.strict == nil {
+		l.strict = MustInterfaceToValue(l.native).(Object)
+	}
+	return l.strict
+}
```

Comment:
Most of the implemented methods below end up forcing the object conversion and calling the method of `ast.Object` -- except for:

- IsGround: an object coming from the store always is
- Len: does not depend on the values
- Find+Get: only converts the requested value
- KeysIterator: converts on demand

Find + Get are a bit special: if the value happens to also be map[string]interface{}, they'll return another lazyObj instead.

Reply:
Nice! I was going to ask about the possible exception cases. :slightly_smiling_face: